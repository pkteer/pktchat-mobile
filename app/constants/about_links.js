// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    TERMS_OF_SERVICE: 'https://mattermost.com/terms-of-use/',
    PRIVACY_POLICY: 'https://mattermost.com/privacy-policy/',
};
