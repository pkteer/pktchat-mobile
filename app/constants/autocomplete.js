// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const AT_MENTION_REGEX = /\B(@([^@\r\n]*))$/i;

export const AT_MENTION_REGEX_GLOBAL = /\B(@([^@\r\n]*))/gi;

export const AT_MENTION_SEARCH_REGEX = /\bfrom:\s*(\S*)$/i;

export const CHANNEL_MENTION_REGEX = /\B(~([^~\r\n]*))$/i;

export const CHANNEL_MENTION_SEARCH_REGEX = /\b(?:in|channel):\s*(\S*)$/i;

export const DATE_MENTION_SEARCH_REGEX = /\b(?:on|before|after):\s*(\S*)$/i;

export const ALL_SEARCH_FLAGS_REGEX = /\b\w+:/g;

export const CODE_REGEX = /(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)| *(`{3,}|~{3,})[ .]*(\S+)? *\n([\s\S]*?\s*)\3 *(?:\n+|$)/g;
