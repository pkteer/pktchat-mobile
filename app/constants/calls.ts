// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const RefreshConfigMillis = 20 * 60 * 1000; // Refresh config after 20 minutes

const RequiredServer = {
    FULL_VERSION: '6.3.0',
    MAJOR_VERSION: 6,
    MIN_VERSION: 3,
    PATCH_VERSION: 0,
};

const PluginId = 'com.mattermost.calls';

// Used for case when cloud server is using Calls v0.5.3.
// This can be removed when v0.5.4 is prepackaged in cloud.
const DefaultCloudMaxParticipants = 8;

export default {RequiredServer, RefreshConfigMillis, PluginId, DefaultCloudMaxParticipants};
