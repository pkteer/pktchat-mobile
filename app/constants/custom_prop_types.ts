// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ReactNode, ReactNodeArray} from 'react';
import {StyleProp, TextStyle, ViewStyle} from 'react-native';

interface CustomPropTypes {
    Children: ReactNode | ReactNodeArray;
    Style: object | number | StyleProp<TextStyle> | Array<StyleProp<TextStyle>> | StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>;
}

export default CustomPropTypes;
