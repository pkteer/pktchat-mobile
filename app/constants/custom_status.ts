// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {CustomStatusDuration} from '@mm-redux/types/users';
import {t} from '@utils/i18n';

const {
    DONT_CLEAR,
    THIRTY_MINUTES,
    ONE_HOUR,
    FOUR_HOURS,
    TODAY,
    THIS_WEEK,
    DATE_AND_TIME,
} = CustomStatusDuration;

export const durationValues = {
    [DONT_CLEAR]: {
        id: t('custom_status.expiry_dropdown.dont_clear'),
        defaultMessage: "Don't clear",
    },
    [THIRTY_MINUTES]: {
        id: t('custom_status.expiry_dropdown.thirty_minutes'),
        defaultMessage: '30 minutes',
    },
    [ONE_HOUR]: {
        id: t('custom_status.expiry_dropdown.one_hour'),
        defaultMessage: '1 hour',
    },
    [FOUR_HOURS]: {
        id: t('custom_status.expiry_dropdown.four_hours'),
        defaultMessage: '4 hours',
    },
    [TODAY]: {
        id: t('custom_status.expiry_dropdown.today'),
        defaultMessage: 'Today',
    },
    [THIS_WEEK]: {
        id: t('custom_status.expiry_dropdown.this_week'),
        defaultMessage: 'This week',
    },
    [DATE_AND_TIME]: {
        id: t('custom_status.expiry_dropdown.date_and_time'),
        defaultMessage: 'Date and Time',
    },
};
export const CUSTOM_STATUS_TEXT_CHARACTER_LIMIT = 100;
export const SET_CUSTOM_STATUS = 'set_custom_status';
