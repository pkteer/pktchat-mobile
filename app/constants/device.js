// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Platform} from 'react-native';
import DeviceInfo from 'react-native-device-info';
import RNFetchBlobFS from 'rn-fetch-blob/fs';

import keyMirror from '@mm-redux/utils/key_mirror';

const deviceTypes = keyMirror({
    CONNECTION_CHANGED: null,
    DEVICE_DIMENSIONS_CHANGED: null,
    DEVICE_TYPE_CHANGED: null,
    DEVICE_ORIENTATION_CHANGED: null,
});

const isPhoneWithInsets = Platform.OS === 'ios' && DeviceInfo.hasNotch();
const isTablet = DeviceInfo.isTablet();
const isIPhone12Mini = DeviceInfo.getModel() === 'iPhone 12 mini';

export default {
    ...deviceTypes,
    DOCUMENTS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Documents`,
    IMAGES_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Images`,
    IS_IPHONE_WITH_INSETS: isPhoneWithInsets,
    IS_TABLET: isTablet,
    VIDEOS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Videos`,
    PERMANENT_SIDEBAR_SETTINGS: '@PERMANENT_SIDEBAR_SETTINGS',
    TABLET_WIDTH: 250,
    AUTOCOMPLETE_MAX_HEIGHT: (isPhoneWithInsets && !isIPhone12Mini) || isTablet ? 200 : 145,
    POST_INPUT_MAX_HEIGHT: (isPhoneWithInsets && !isIPhone12Mini) || isTablet ? 150 : 88,
};
