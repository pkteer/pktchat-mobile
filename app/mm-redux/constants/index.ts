// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import Emoji from './emoji';
import Files from './files';
import General from './general';
import Groups from './groups';
import Permissions from './permissions';
import Plugins from './plugins';
import Posts from './posts';
import Preferences from './preferences';
import RequestStatus from './request_status';
import Roles from './roles';
import Stats from './stats';
import Teams from './teams';
import Users from './users';

export {General, Preferences, Posts, Files, RequestStatus, Teams, Stats, Permissions, Emoji, Plugins, Groups, Users, Roles};
