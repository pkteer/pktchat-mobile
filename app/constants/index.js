// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import AttachmentTypes from './attachment';
import CustomPropTypes from './custom_prop_types';
import * as CustomStatus from './custom_status';
import DateTypes from './date';
import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import Sso from './sso';
import Types from './types';
import ViewTypes from './view';
import WebsocketEvents from './websocket';

export {
    AttachmentTypes,
    CustomPropTypes,
    DeepLinkTypes,
    DeviceTypes,
    DateTypes,
    ListTypes,
    NavigationTypes,
    Sso,
    Types,
    ViewTypes,
    WebsocketEvents,
    CustomStatus,
};
