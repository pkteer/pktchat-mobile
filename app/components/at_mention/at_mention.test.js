// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import {Preferences} from '@mm-redux/constants';

import AtMention from './at_mention.js';

describe('AtMention', () => {
    const baseProps = {
        usersByUsername: {},
        mentionKeys: [{key: 'John.Smith'}, {key: 'Jane.Doe'}],
        teammateNameDisplay: '',
        mentionName: 'John.Smith',
        mentionStyle: {color: '#ff0000'},
        textStyle: {backgroundColor: 'yellow'},
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot, no highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, with highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        wrapper.setState({user: {username: 'John.Smith'}});
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, without highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        wrapper.setState({user: {username: 'Victor.Welch'}});
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
