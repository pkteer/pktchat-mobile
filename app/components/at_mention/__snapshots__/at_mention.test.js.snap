// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtMention should match snapshot, no highlight 1`] = `
<Text
  style={
    Object {
      "backgroundColor": "yellow",
    }
  }
>
  <Text
    style={Array []}
  >
    @John.Smith
  </Text>
</Text>
`;

exports[`AtMention should match snapshot, with highlight 1`] = `
<Text
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "yellow",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ff0000",
        },
        Object {
          "backgroundColor": "#ffd470",
          "color": "#1b1d22",
        },
      ]
    }
  >
    @John.Smith
  </Text>
</Text>
`;

exports[`AtMention should match snapshot, without highlight 1`] = `
<Text
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "yellow",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ff0000",
        },
      ]
    }
  >
    @Victor.Welch
  </Text>
</Text>
`;
